### PR TITLE
LPD-30624 Embedded portlets with persistSettings disabled don't consider javascript in footer and header

### PR DESCRIPTION
**Pull request by dsanz**
In this PR, we are:

- Making RuntimeTag emit portlet header and footer JS/CSS resource links as long as portlet is rendered
- Avoiding repetition of such resource links in case portlet is rendered more than once in the same page. 
- Adding integration tests

Additional notes:
 - Header/Footer CSS/JS links are rendered regardless the existence/count of portlet preferences and `persistSettings` flag. 
 - Exception to the above is when portlet has been already rendered in the page. This PR handles that case also:
   - Prior to this, embedded portlet render logic was not able to identify whether the header/footer resoures were already visited (emitted) or not
   - PR leverages existing `PortletRenderUtil` logic that prevents repeated links from being emitted, by turning a request attribute into a shared one so that it'll be found from different request contexts, included those of `RuntimeTag` 
 - I've discarded other custom solutions such as thread-locals as there is already a piece of logic handling this repetition, and thread-locals require cleanup after request is served.
 - Common limitations to dynamically added regular portlet exists with embedded: when a new portlet is added to a page, it's rendered in its own, isolated context, so the header/footer CSS/JS will be emitted. If the added portlet uses `RuntimeTag` to render another portlet, then, this embedded portlet will also emit the headers/footers

### CODE REVIEW

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
I'm waiting till SF rules get an update @bryceosterhaus you're working on this SF now right?

**Comment by dsanz** [comment]
Force pushed in the hope SF issues are gone

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
ci:test:sf

**Comment by dsanz** [comment]
ci:test:relevant

File: portal-kernel/src/com/liferay/portal/kernel/util/WebKeys.java [hunk 1]
```diff
@@ -495,7 +495,7 @@ public interface WebKeys {
 		"PORTLET_RENDER_PARAMETERS_";
 
 	public static final String PORTLET_RESOURCE_STATIC_URLS =
-		"PORTLET_RESOURCE_STATIC_URLS";
+		"LIFERAY_SHARED_PORTLET_RESOURCE_STATIC_URLS";
```
Comment (jkappler) [inline portal-kernel/src/com/liferay/portal/kernel/util/WebKeys.java:498]: @dsanz, BChan will most likely ask for a change of variable, Or a new variable in any case...

**Comment by dsanz** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
@jkappler this passes the tests in the CI. For some reason, they don't pass locally, however I believe this is not a reason to hold this pull request.

I'll let Brian decide about variable naming as I've seen [changes like this](https://github.com/liferay-frontend/liferay-portal/pull/4350/commits/f21dbdffb46d91320ee72f28ac9eaa910cec482b) in the past. If he ends up rejecting we'll think a different solution

Thank you for your effort!

**Comment by dsanz** [comment]
ci:forward:force

**Pull request closed without merging**